examples/MQ/9-PixelDetector/src/devices/FairMQSimDevice.h

```diff
@@ -0,0 +1,107 @@
+/********************************************************************************
+ *    Copyright (C) 2017 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             * 
+ *         GNU Lesser General Public Licence version 3 (LGPL) version 3,        *  
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+/**
+ * FairMQSimDevice.h
+ *
+ * @since 2018-02-01
+ * @author R. Karabowicz
+ */
+
+#ifndef FAIRMQSIMDEVICE_H_
+#define FAIRMQSIMDEVICE_H_
+
+#include <string>
+
+#include "FairSink.h"
+#include "FairMQDevice.h"
+
+class FairEventHeader;
+class FairRunSim;
+class TObjArray;
+class FairField;
+class FairParIo;
+class FairPrimaryGenerator;
+
+class FairMQSimDevice : public FairMQDevice, public FairSink
```

Comment:
Is a double inheritance needed? Is a device simulation really a particular specialization of a FairSink?
(To me this looks at least a bit odd).

Reply:
I will try to answer the comments in one:

1. Alexey, FairLogger::endl will go;
2. Sandro: TObject inheritance is probably not needed. I will remove it.
3. I will replace Init() by InitSink().
4. The PR changes RootManager - I agree, that more test are needed.
5. Double inheritance was an initial idea of mine and it works. The FairMQSimDevice is both the simulation data sink and the device that sends the data.